perfkitbenchmarker/disk.py

```
@@ -26,10 +26,29 @@
 PIOPS = 'piops'  # Provisioned IOPS (SSD) in AWS
 LOCAL = 'local'
 
+_DISK_SPEC_REGISTRY = {}
+
+
+def GetDiskSpecClass(cloud):
+  """Get the DiskSpec class corresponding to 'cloud'."""
+  return _DISK_SPEC_REGISTRY.get(cloud, BaseDiskSpec)
+
+
+class AutoRegisterDiskSpecMeta(type):
+  """Metaclass which automatically registers DiskSpecs."""
+
+  def __init__(cls, name, bases, dct):
+    if cls.CLOUD:
```

Should we raise an exception if cls.CLOUD is None? That seems like an almost-certain indication of a bug (except possibly for static VMs).